components/clarity-repl/src/repl/datastore.rs

```diff
@@ -98,107 +95,117 @@ fn height_to_id(height: u32) -> StacksBlockId {
+            block_id_lookup: HashMap::from([(id, id)]),
+            height_at_chain_tip: HashMap::from([(id, 0)]),
         }
     }
 
-    pub fn advance_chain_tip(&mut self, count: u32) -> u32 {
-        let cur_height = self.chain_height;
-        let current_lookup_id = *self
+    pub fn open(_path_str: &str, _miner_tip: Option<&StacksBlockId>) -> Result<ClarityDatastore> {
+        Ok(ClarityDatastore::new())
+    }
+
+    pub fn as_analysis_db(&mut self) -> AnalysisDatabase<'_> {
+        AnalysisDatabase::new(self)
+    }
+
+    /// begin, commit, rollback a save point identified by key
+    ///    this is used to clean up any data from aborted blocks
+    ///     (NOT aborted transactions that is handled by the clarity vm directly).
+    /// The block header hash is used for identifying savepoints.
+    ///     this _cannot_ be used to rollback to arbitrary prior block hash, because that
+    ///     blockhash would already have committed and no longer exist in the save point stack.
+    /// this is a "lower-level" rollback than the roll backs performed in
+    ///   ClarityDatabase or AnalysisDatabase -- this is done at the backing store level.
+
+    pub fn begin(&mut self, _current: &StacksBlockId, _next: &StacksBlockId) {
+        // self.marf.begin(current, next)
+        //     .expect(&format!("ERROR: Failed to begin new MARF block {} - {})", current, next));
+        // self.chain_tip = self.marf.get_open_chain_tip()
+        //     .expect("ERROR: Failed to get open MARF")
+        //     .clone();
+        // self.side_store.begin(&self.chain_tip);
+    }
+    pub fn rollback(&mut self) {
+        // self.marf.drop_current();
+        // self.side_store.rollback(&self.chain_tip);
+        // self.chain_tip = StacksBlockId::sentinel();
+    }
+    // This is used by miners
+    //   so that the block validation and processing logic doesn't
+    //   reprocess the same data as if it were already loaded
+    pub fn commit_mined_block(&mut self, _will_move_to: &StacksBlockId) {
+        // rollback the side_store
+        //    the side_store shouldn't commit data for blocks that won't be
+        //    included in the processed chainstate (like a block constructed during mining)
+        //    _if_ for some reason, we do want to be able to access that mined chain state in the future,
+        //    we should probably commit the data to a different table which does not have uniqueness constraints.
+        // self.side_store.rollback(&self.chain_tip);
+        // self.marf.commit_mined(will_move_to)
+        //     .expect("ERROR: Failed to commit MARF block");
+    }
+
+    pub fn commit_to(&mut self, _final_bhh: &StacksBlockId) {
+        // println!("commit_to({})", final_bhh);
+        // self.side_store.commit_metadata_to(&self.chain_tip, final_bhh);
+        // self.side_store.commit(&self.chain_tip);
+        // self.marf.commit_to(final_bhh)
+        //     .expect("ERROR: Failed to commit MARF block");
+    }
```

Comment:
Why is the code in this section commented out?

Reply:
nvm saw your comment below. in general i prefer deleting code that is not used to reduce cognitive overhead. It's still in the history if needed.

Reply:
Yes I agree, but I'm not sure why it was left there in the first. And since it's not really the point of this PR, I'll come back at it later I think